src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/JsonObjectConverter.cs

```diff
@@ -9,7 +9,18 @@ namespace System.Text.Json.Serialization
     /// </summary>
     internal abstract class JsonObjectConverter<T> : JsonResumableConverter<T>
     {
+        internal JsonObjectConverter()
+        {
+            // Populate ElementType if the runtime type implements IAsyncEnumerable.
+            // Used to feed the (converter-agnostic) JsonClassInfo.ElementType instace
+            // which is subsequently consulted by custom enumerable converters fed via JsonConverterAttribute.
+            if (IAsyncEnumerableConverterFactory.TryGetAsyncEnumerableInterface(typeof(T), out Type? asyncEnumerableInterface))
+            {
+                ElementType = asyncEnumerableInterface.GetGenericArguments()[0];
```

Comment:
In which scenario is the element type needed when the type is being treated as `ClassType.Object`?

Reply:
This is related to https://github.com/dotnet/runtime/issues/50205. But that being said, making `IAsyncEnumerableConverter` the default behaviour (vis a vis the original proposed behaviour of opting into it) means that this workaround is no longer needed (in this context). I'll revert the changes.